api.js

```diff
@@ -22,20 +19,28 @@ function init(page, observer, prevSpeed) {
 		}
 
 		if (result.isDone) {
-			page.close();
+			browser.close();
 			observer.complete();
 		} else {
-			setTimeout(init, 100, page, observer, result.speed);
+			setTimeout(init, 100, page, browser, observer, result.speed);
 		}
 	})
 	.catch(err => observer.error(err));
 }
 
 module.exports = () => new Observable(observer => {
-	driver.create({path: phantomjs.path})
-		.then(phantom => phantom.createPage())
-		.then(page => page.open('http://fast.com').then(() => {
-			init(page, observer);
-		}))
+	let browserInstance;
+	let pageInstance;
+
+	puppeteer.launch()
+		.then(browser => {
+			browserInstance = browser;
+			return browserInstance.newPage();
+		})
+		.then(page => {
+			pageInstance = page;
+			return pageInstance.goto('http://fast.com');
+		})
+		.then(() => init(pageInstance, browserInstance, observer))
```

Comment:
You can do this instead of having to reassign `browserInstance` and `pageInstance`:

```js
puppeteer.launch()
		.then(browser => Promise.all([browser, browser.newPage()]))
		.then(([browser, page]) => Promise.all([browser, page, page.goto('http://fast.com')]))
		.then(([browser, page]) => init(pageInstance, browserInstance, observer))
		.catch(err => observer.error(err));
```

Reply:
Thanks for sharing, definitely a cleaner pattern. Will push the refactor. Would be a little uglier if 4 support is maintained since we lose array destructuring.